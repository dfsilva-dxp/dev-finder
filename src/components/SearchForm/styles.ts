import styled, { css } from "styled-components";

export const Form = styled.form`
  ${({ theme }) => css`
    padding: ${theme.spacings.xxsmall};
    border-radius: ${theme.border.radius};
    background-color: ${theme.colors.panel};
    outline-offset: ${theme.border.outlineOffiset};
    border: 2px solid transparent;

    display: flex;
    align-items: center;
    gap: ${theme.spacings.xxsmall};

    &:focus-within {
      outline-offset: ${theme.border.outlineOffiset};
      border: 2px solid ${theme.colors.primary};
    }

    input[type="search"] {
      flex: 1;
      padding: ${theme.spacings.xxsmall} ${theme.spacings.xsmall};
      background-color: transparent;
      border: 0;
      outline: none;
    }

    button[type="submit"] {
      padding: ${theme.spacings.xxsmall} ${theme.spacings.xsmall};
      border-radius: ${theme.border.radius};
      border: 0;
      box-shadow: 0;
      outline: 0;
      outline-offset: ${theme.border.outlineOffiset};
      border: 2px solid transparent;
      background-color: ${theme.colors.primary};
      color: #fff;
      font-weight: ${theme.font.weight.bold};

      &:hover {
        filter: brightness(0.9);
      }

      &:focus {
        outline-offset: ${theme.border.outlineOffiset};
        border: 2px solid #fff;
      }
    }

    svg {
      color: ${theme.colors.primary};
    }
  `}
`;
