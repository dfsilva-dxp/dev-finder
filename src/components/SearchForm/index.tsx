import { MagnifyingGlass } from "phosphor-react";
import { ChangeEvent, FormEvent, useState } from "react";

import useGithubUser from "hooks/useGithubUser";

import { Form } from "./styles";

export default function SearchForm() {
  const [value, setValue] = useState("");

  const { searchGithubUser } = useGithubUser();

  const onChange = (evt: ChangeEvent<HTMLInputElement>) => {
    setValue(evt.target.value);
  };

  const handleSearchUser = (evt: FormEvent<HTMLFormElement>) => {
    evt.preventDefault();

    searchGithubUser(value);
    setValue("");
  };

  return (
    <Form onSubmit={handleSearchUser} aria-label="Search a GitHub username">
      <MagnifyingGlass size={20} />
      <input
        type="search"
        placeholder="Search GitHub username..."
        onChange={onChange}
        value={value}
      />
      <button type="submit">Search</button>
    </Form>
  );
}
