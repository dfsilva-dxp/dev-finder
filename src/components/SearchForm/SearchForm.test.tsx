import { screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { renderWithTheme } from "utils/test/helpers";

import SearchForm from ".";

describe("<SearchForm", () => {
  it("should renders with placeholder", () => {
    renderWithTheme(<SearchForm />);

    expect(
      screen.getByPlaceholderText(/search github username.../i)
    ).toBeInTheDocument();
  });

  it("should changes its value when typing", async () => {
    const { container } = renderWithTheme(<SearchForm />);

    const input = screen.getByPlaceholderText(/search github username.../i);
    const text = "This is my new text";

    userEvent.type(input, text);

    await waitFor(() => {
      expect(input).toHaveValue(text);
    });

    expect(container).toMatchSnapshot();
  });
});
