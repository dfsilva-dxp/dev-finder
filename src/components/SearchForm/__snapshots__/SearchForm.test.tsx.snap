// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm should changes its value when typing 1`] = `
.c0 {
  padding: 0.5rem;
  border-radius: 0.25rem;
  background-color: #393656;
  outline-offset: 0.125rem;
  border: 2px solid transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
}

.c0:focus-within {
  outline-offset: 0.125rem;
  border: 2px solid #FE5873;
}

.c0 input[type="search"] {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0.5rem 1rem;
  background-color: transparent;
  border: 0;
  outline: none;
}

.c0 button[type="submit"] {
  padding: 0.5rem 1rem;
  border-radius: 0.25rem;
  border: 0;
  box-shadow: 0;
  outline: 0;
  outline-offset: 0.125rem;
  border: 2px solid transparent;
  background-color: #FE5873;
  color: #fff;
  font-weight: 700;
}

.c0 button[type="submit"]:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c0 button[type="submit"]:focus {
  outline-offset: 0.125rem;
  border: 2px solid #fff;
}

.c0 svg {
  color: #FE5873;
}

<div>
  <form
    aria-label="Search a GitHub username"
    class="c0"
  >
    <svg
      fill="currentColor"
      height="20"
      viewBox="0 0 256 256"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="none"
        height="256"
        width="256"
      />
      <circle
        cx="116"
        cy="116"
        fill="none"
        r="84"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
      />
      <line
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
        x1="175.4"
        x2="224"
        y1="175.4"
        y2="224"
      />
    </svg>
    <input
      placeholder="Search GitHub username..."
      type="search"
      value="This is my new text"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
